sstables/sstable_set.cc

```diff
@@ -182,6 +182,34 @@ sstable_set::erase(shared_sstable sst) {
     return _impl->erase(sst);
 }
 
+// Default implementation for sstable_set_impl vector insert
+std::vector<shared_sstable>
+sstable_set_impl::insert(const std::span<shared_sstable>& sstables) {
+    std::vector<shared_sstable> inserted;
+    inserted.reserve(sstables.size());
+    try {
+        for (const auto& sst : sstables) {
+            if (insert(sst)) {
+                inserted.emplace_back(sst);
+            }
+        }
+    } catch (...) {
+        for (const auto& sst : inserted) {
+            auto erased = erase(sst);
+            if (!erased) {
+                on_fatal_internal_error(sstlog, fmt::format("sstable {} was not erased after insert failure: {}", sst, std::current_exception()));
+            }
+        }
+        throw;
+    }
+    return inserted;
+}
```

Comment:
we don't need it at sstable_set_impl. this is only a helper which reuses existing methods. for compound set, the first insertion attempt will trigger the exception.

Reply:
Good point.  Moves to sstable_set in the next rev.